File: de.unistuttgart.iste.sqa.mpw.karasimulator.cpp/core/src/impl/GameLogImpl.cpp
#include "GameLogImpl.h"

#include <stdexcept>

namespace mpw {

GameLogImpl::GameLogImpl() {
}

void GameLogImpl::write(std::shared_ptr<mpw::Actor> actor,
		std::string message) {
	throw std::runtime_error("not implemented");
}

}

File: simulators/de.unistuttgart.iste.sqa.mpw.framework.java/core/src/main/java/de/unistuttgart/iste/sqa/mpw/framework/mpw/impl/GameLogImpl.java
package de.unistuttgart.iste.sqa.mpw.framework.mpw.impl;

import de.unistuttgart.iste.sqa.mpw.framework.mpw.Actor;
import de.unistuttgart.iste.sqa.mpw.framework.mpw.GameLog;
import de.unistuttgart.iste.sqa.mpw.framework.mpw.LogEntry;

public class GameLogImpl extends GameLog {

	@Override
	public void write(final Actor actor, final String message) {
		final var logEntry = new LogEntry();
		logEntry.setActor(actor);
		logEntry.setMessage(message);
		addToLogEntries(logEntry);
	}
}
